src/tensor_product_basis.jl

```diff
@@ -0,0 +1,42 @@
+abstract type TensorProductBasis <: Function end
+
+##TODO: Chebyshev Basis
+
+##Fourier basis of the form F_j(x) = j is odd ? cos((j÷2)*x) : sin((j÷2)*x) => [F_0(x), F_1(x), ..., F_n(x)]
+struct FourierBasis{Int} <: TensorProductBasis
+    n::Int
+
+    function FourierBasis(n)
+        new{typeof(n)}(n)
+    end
+end
+
+function (basis::FourierBasis)(x)
+    return Fourier()[x, 1:basis.n]
```

Comment:
From where are you using the `Fourier`?

Reply:
@Vaibhavdixit02 from `OrthogonalPolynomialsQuasi.jl`.